tests/state/test_get_states.py

```diff
@@ -0,0 +1,346 @@
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--address',
+        non_existing_address,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_error_message = {
+        'errors': f'Block not found.',
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error_message) in result.output
+
+
+def test_get_states_with_limit():
+    """
+    Case: get a list of states by limit.
+    Expect: list of states is returned.
+    """
+    limit = 1
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--limit',
+        limit,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    single_state_result = json.loads(result.output).get('result')
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert len(single_state_result) == limit
+
+
+def test_get_states_with_invalid_limit():
+    """
+    Case: get a list of states by its invalid limit.
+    Expect: the following limit is invalid error message.
+    """
+    invalid_limit = '-3'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--limit',
+        invalid_limit,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_error_message = {
+        'errors': {
+            'limit': [
+                'Limit must be greater than 0.',
+            ],
+        },
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error_message) in result.output
+
+
+def test_get_states_with_non_existing_limit():
+    """
+    Case: get a list of states by its non-existing limit.
+    Expect: invalid limit count error message.
+    """
+    non_existing_limit = 1e+10**2
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--limit',
+        non_existing_limit,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_error_message = {
+        'errors': 'Invalid limit count.',
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error_message) in result.output
+
+
+def test_get_states_with_reverse():
+    """
+    Case: get a list of states by reverse.
+    Expect: reverse list of states is returned.
+    """
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--reverse',
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    result_reverse_states = json.loads(result.output).get('result')
+    first_reverse_state = result_reverse_states[0]
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert first_reverse_state
```

Comment:
Check a single's state address (account address regexp) instead of just existing.

Reply:
@dmytrostriletskyi, done.